internal/configs/configschema/path.go

```diff
@@ -58,3 +58,20 @@ func (o *Object) AttributeByPath(path cty.Path) *Attribute {
 	}
 	return nil
 }
+
+// PathName is trying to get the name of the attribute referenced by the cty.Path.
+// Due to the implementation difference between cty.IndexStep and cty.GetAttrStep,
+// this method is trying to find the most right-sided cty.GetAttrStep and returns the
+// name of that.
+// If the path contains no cty.GetAttrStep, it returns <unknown>.
+// TODO ephemeral - is there a better way to do this?
```

Comment:
Only other option I can think of is a recursive call, which does not improve over what you have written here.